### PR TITLE
feat(events): support customer managed key for event bus

### PR DESCRIPTION
**Pull request by mazyu36**
### Issue # (if applicable)
N/A

### Reason for this change
Amazon EventBridge supports Customer Managed Key for Event Bus since 2024/5.
But current L2 Construct does not support CMK.

For more information, see [Amazon EventBridge now supports Customer Managed Keys (CMK) for Event Buses](https://aws.amazon.com/about-aws/whats-new/2024/05/amazon-eventbridge-cmk-event-buses/?nc1=h_ls).

### Description of changes
Add `kmsKey` property to the `EventBus` class

### Description of how you validated changes
Add unit tests and integ tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-events/README.md
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-events/README.md:253]:
```suggestion
## Use a customer managed key

To use a customer managed key for events on the event bus, use the `kmsKey` attribute.
```
Spacing

File: packages/aws-cdk-lib/aws-events/README.md
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-events/README.md:263]: Might be useful to add an extra note about archives and schema discovery not working with CMK ([docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-events-eventbus.html#cfn-events-eventbus-kmskeyidentifier)).

File: packages/aws-cdk-lib/aws-events/test/event-bus.test.ts
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-events/test/event-bus.test.ts:644]: Can you please also add an assertion on the key policy?
```ts
Template.fromStack(stack).hasResourceProperties('AWS::KMS::Key', {
          KeyPolicy: {
            'Statement': [ ... ]
          }
})
```

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks 👍 Left suggestions for minor improvements

File: packages/aws-cdk-lib/aws-events/README.md
Comment (mazyu36) [reply packages/aws-cdk-lib/aws-events/README.md:264]: Thanks.
I added the note.

File: packages/aws-cdk-lib/aws-events/test/event-bus.test.ts
Comment (mazyu36) [reply packages/aws-cdk-lib/aws-events/test/event-bus.test.ts:644]: Thanks.
I added an assertion for the key policy.

**Comment by mazyu36** [comment]
@lpizzinidev 
Thank you for the review!
I'v addressed all your comments.

File: packages/aws-cdk-lib/aws-events/README.md
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-events/README.md:267]:
```suggestion
To enable archives or schema discovery on an event bus, choose to use an AWS owned key.
For more information, see [Data encryption for EventBridge event buses](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-encryption-event-bus.html).
```

**Review by lpizzinidev — APPROVED** [review]
Thanks 👍

File: packages/aws-cdk-lib/aws-events/lib/event-bus.ts
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-events/lib/event-bus.ts:344]: The resource can be more strict on the event bus right?

**Review by GavinZZ — CHANGES_REQUESTED** [review]
Thanks for the PR, two comments to address before I approve.

File: packages/aws-cdk-lib/aws-events/README.md
Comment (mazyu36) [reply packages/aws-cdk-lib/aws-events/README.md]: Thanks. I've fixed it.
Upon checking, it seems that the document structure and links have changed while waiting for the review.

File: packages/aws-cdk-lib/aws-events/lib/event-bus.ts
Comment (mazyu36) [reply packages/aws-cdk-lib/aws-events/lib/event-bus.ts:344]: I think this wild card is correct.
This key policy complies with the one specified in the event-bus [documentation](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-encryption-event-bus-key-policy.html).


The wild card (*) in the Resource of the Key policy represents "this key" (Ref [docs](https://docs.aws.amazon.com/kms/latest/developerguide/key-policy-overview.html#key-policy-elements)).

> (Required) In a key policy, the value of the Resource element is "*", which means "this KMS key." The asterisk ("*") identifies the KMS key to which the key policy is attached.

For the above reason, it is common to specify a wildcard in the resource of the key policy. For example, it is also specified in the following:

https://github.com/aws/aws-cdk/blob/8651bbe57d77e2e4a4ae70f7abb1c8af42687058/packages/aws-cdk-lib/aws-sns-subscriptions/lib/sqs.ts#L68

Comment (GavinZZ) [reply packages/aws-cdk-lib/aws-events/lib/event-bus.ts:344]: That's good to know and thanks for the update!

**Review by GavinZZ — APPROVED** [review]
LGTM, thank you!

**Comment by mazyu36** [comment]
@Mergifyio update

**Comment by mazyu36** [comment]
@GavinZZ 
Thank you for the review. Could you please approve it again?

I merged main because the following error appeared:
`This pull request cannot be embarked for merge​​​​​​​​​​​​​​​​`

**Review by GavinZZ — APPROVED** [review]

**Pull request merged**
